packages/@aws-cdk/aws-stepfunctions-tasks/lib/athena/start-query-execution.ts

```diff
@@ -0,0 +1,289 @@
+export interface AthenaStartQueryExecutionProps extends sfn.TaskStateBaseProps {
+  /**
+   * Query that will be started
+   */
+  readonly queryString: string;
+
+  /**
+   * Unique string string to ensure idempotence
+   *
+   * @default - No client request token
+   */
+  readonly clientRequestToken?: string;
+
+  /**
+   * Database within which query executes
+   *
+   * @default - No query execution context
+   */
+  readonly queryExecutionContext?: QueryExecutionContext;
+
+  /**
+   * Configuration on how and where to save query
+   *
+   * @default - No result configuration
+   */
+  readonly resultConfiguration?: ResultConfiguration;
+
+  /**
+   * Configuration on how and where to save query
+   *
+   * @default - No work group
+   */
+  readonly workGroup?: string;
+}
+
+/**
+ * Start an Athena Query as a Task
+ *
+ * @see https://docs.aws.amazon.com/step-functions/latest/dg/connect-athena.html
+ * @experimental
+ */
+export class AthenaStartQueryExecution extends sfn.TaskStateBase {
+
+  private static readonly SUPPORTED_INTEGRATION_PATTERNS: sfn.IntegrationPattern[] = [
+    sfn.IntegrationPattern.REQUEST_RESPONSE,
+    sfn.IntegrationPattern.WAIT_FOR_TASK_TOKEN,
+  ];
+
+  protected readonly taskMetrics?: sfn.TaskMetricsConfig;
+  protected readonly taskPolicies?: iam.PolicyStatement[];
+
+  private readonly integrationPattern: sfn.IntegrationPattern;
+
+  constructor(scope: Construct, id: string, private readonly props: AthenaStartQueryExecutionProps) {
+    super(scope, id, props);
+    this.integrationPattern = props.integrationPattern ?? sfn.IntegrationPattern.REQUEST_RESPONSE;
+
+    validatePatternSupported(this.integrationPattern, AthenaStartQueryExecution.SUPPORTED_INTEGRATION_PATTERNS);
+
+    this.taskPolicies = this.createPolicyStatements();
+  }
+
+  private createPolicyStatements(): iam.PolicyStatement[] {
+    const policyStatements = [
+      new iam.PolicyStatement({
+        resources: [
+          cdk.Stack.of(this).formatArn({
+            service: 'athena',
+            resource: 'datacatalog',
+            resourceName: this.props.queryExecutionContext?.catalogName ?? 'AwsDataCatalog',
+          }),
+          cdk.Stack.of(this).formatArn({
+            service: 'athena',
+            resource: 'workgroup',
+            resourceName: this.props.workGroup ?? 'primary',
+          }),
+
+        ],
+        actions: ['athena:getDataCatalog', 'athena:startQueryExecution'],
+      }),
+    ];
+
+    policyStatements.push(
+      new iam.PolicyStatement({
+        actions: ['s3:AbortMultipartUpload',
+          's3:CreateBucket',
+          's3:GetBucketLocation',
+          's3:GetObject',
+          's3:ListBucket',
+          's3:ListBucketMultipartUploads',
+          's3:ListMultipartUploadParts',
+          's3:PutObject'],
+        resources: [this.props.resultConfiguration?.outputLocation ?? '*'], // Need S3 location where data is stored https://docs.aws.amazon.com/athena/latest/ug/security-iam-athena.html
```

Comment:
Since these policy resources expect ARN, passing `outputLocation` (S3 URL like `s3://...`) will get an error.
Even if `outputLocation` is specified to props, we may have to pass `*` to resources or parse S3 URL to get ARN correctly.

Reply:
Thank you for spotting this. I parsed the s3 bucket correctly like you suggested but still got an error `Unable to verify/create output bucket`. When I reran the exact same state machine but with wildcard permissions the machine succeeded and sent the logs to the correct output location. I will be submitting a fix immediately granting wildcard permissions for S3 and LakeFormation